src/react-components/src/components/01-atoms/Button/Button.tsx

```diff
@@ -4,10 +4,37 @@ import bem from "../../../utils/bem";
 
 type BtnContent = string | JSX.Element;
 
+export interface ButtonOptions {
+  id: string;
+
+  /** The action to perform on the <button>'s onClick function */
+  callback?: (event: React.MouseEvent | React.KeyboardEvent) => void;
+  /** The content to render inside the the button. An alternative
+   * to passing children elements. */
+  content?: string | JSX.Element;
+  attributes?: {};
+  /** Used for BEM css convention. */
+  modifiers?: string[];
+  /** Used for BEM css convention. */
+  blockName?: string;
+  large?: boolean;
+  type?: string;
+  /** The html button attribute */
+  buttonType?: string;
+  mouseDown?: boolean;
+  /** If an icon is to be rendered, an `iconPosition` prop is required. */
+  iconPosition?: string;
+  iconName?: string;
```

Comment:
I would like to get annotations for all of these as we work on them since I think down the road the larger UX team (and other developers) will be able to click the "More Info" link and see all the available options on a component (particularly useful for IA/Ellen). Even if it feels redundant to us, a description on each of these props will be helpful to others.

Reply:
yeah definitely.  There's a lot of stuff that doesn't have them but new stuff/stuff that's being updated should have them for sure.